app/helpers/flyover_comments/comments_helper.rb

```diff
@@ -115,12 +115,12 @@ def flag_flyover_comment_modal
       render "flyover_comments/flags/modal"
     end
 
-    def user_already_flagged_comment?(comment, user = _flyover_comments_current_user)
-      FlyoverComments::Flag.where(:comment => comment, FlyoverComments.user_class_symbol => user).exists?
+    def user_already_flagged_comment?(comment, user = comment.commenter)
```

Comment:
This I think we also need to stick to `_flyover_comments_current_user`. This method is figuring out whether the currently logged in user has flagged a comment or not. It could very well be someone other than the original commenter.